src/components/molecules/AuthForm/AuthForm.tsx

```diff
@@ -123,13 +124,17 @@ const AuthForm: FC<AuthProps> = ({ type }) => {
           fullWidth
           onClick={type === EAuthType.Login ? handleLogin : handleSignup}
         >
-          {type === EAuthType.Signup
-            ? loading
-              ? 'loading...'
-              : 'Create an account'
-            : loading
-            ? 'loading...'
-            : 'Login to your account'}
+          {type === EAuthType.Signup ? (
+            loading ? (
+              <Spinner size={20} />
+            ) : (
+              'Create an account'
+            )
+          ) : loading ? (
+            <Spinner size={20} />
+          ) : (
+            'Login to your account'
+          )}
```

Comment:
I think this does the same with a bit less repetition:
```suggestion
          {loading ? 
              <Spinner size={20} />
            ) :  type === EAuthType.Signup ? (
              'Create an account'
            ) : (
              'Login to your account'
          )}
```

Alternatively, you could create a helper function `getSubmitButtonContent`:
```
    const getSubmitButtonContent = () => {
        if (loading) return <Spinner size={20} />
        
        return type === EAuthType.Signup ? 'Create an account' : 'Login to you account' 
    }
```